app/subscriber/src/features/my-reports/edit/ReportEditContext.tsx

```diff
@@ -0,0 +1,144 @@
+import { useFormikContext } from 'formik';
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { IReportInstanceModel } from 'tno-core';
+
+import { defaultReport } from '../constants';
+import { IReportForm, IReportInstanceContentForm } from '../interfaces';
+import {
+  ReportContentMenuOption,
+  ReportMainMenuOption,
+  ReportPreviewMenuOption,
+  ReportSettingsMenuOption,
+} from './constants';
+
+/*******************************************************************
+ * All related chart template context code is in a single file because it will never be used separately.
+ *******************************************************************/
+
+/**
+ * IReportEditContext interface for the context.
+ */
+export interface IReportEditContext {
+  /** The active report page */
+  active?: string;
+  /** Change the active report page. */
+  setActive: React.Dispatch<React.SetStateAction<string | undefined>>;
+  /** The active content row. */
+  activeRow?: IReportInstanceContentForm;
+  /** Change the active content row. */
+  setActiveRow: React.Dispatch<React.SetStateAction<IReportInstanceContentForm | undefined>>;
+  /** Formik values. */
+  values: IReportForm;
+  /** Formik setFieldValue function */
+  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+  /** Formik setValues function */
+  setValues: (
+    values: React.SetStateAction<IReportForm>,
+    shouldValidate?: boolean | undefined,
+  ) => void;
+  isSubmitting: boolean;
+  setSubmitting: (isSubmitting: boolean) => void;
+  submitForm: (() => Promise<void>) & (() => Promise<any>);
+  handleNavigate: (instance: IReportInstanceModel | undefined, action: 'previous' | 'next') => void;
+}
+
+/**
+ * ReportEditContext provides a shared context for chart template components.
+ */
+export const ReportEditContext = React.createContext<IReportEditContext>({
```

Comment:
This is the start of extracting functionality away from components to simplify the report forms.